packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts

```diff
@@ -177,18 +192,85 @@ export class AzureCommunicationCallWithChatAdapter implements CallWithChatAdapte
     };
 
     this.callAdapter.onStateChange(onCallStateChange);
+    /* @conditional-compile-remove(breakout-rooms) */
+    this.callAdapter.on('breakoutRoomsUpdated', async (eventData: BreakoutRoomsEventData) => {
+      if (!eventData.data) {
+        return;
+      }
+      if (eventData.type === 'join') {
+        await this.breakoutRoomJoined(eventData.data);
+      } else if (eventData.type === 'assignedBreakoutRooms') {
+        if (eventData.data.state === 'closed') {
+          this.returnFromBreakoutRoom();
+        }
+      }
+    });
     this.onCallStateChange = onCallStateChange;
   }
 
+  /* @conditional-compile-remove(breakout-rooms) */
+  private async breakoutRoomJoined(call: Call | TeamsCall): Promise<void> {
+    const targetThreadId = call.info.threadId;
+    // If the chat adapter is not on the target thread then we need to switch to the breakout room chat adapter
+    if (targetThreadId && this.chatAdapter && this.chatAdapter.getState().thread.threadId !== targetThreadId) {
```

Comment:
(reviewer todo: adding comment to remind me that I want to review if there's any race conditions in the existing logic that waits for threadId to come in from the call object)